employees/tests/test_views.py

```diff
@@ -107,6 +113,87 @@ def test_admin_report_detail_view_should_update_report_on_post(self):
+            "custom-report-list",
+            kwargs={"year": datetime.datetime.now().date().year, "month": datetime.datetime.now().date().month},
+        )
+        self.data = {
+            "date": datetime.datetime.now().date(),
+            "description": "Some description",
+            "project": self.report.project.pk,
+            "author": self.user.pk,
+            "work_hours": "8:00",
+            "task_activities": self.report.task_activities.pk,
+        }
+
+    def test_custom_list_view_should_display_users_report_list_on_get(self):
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, self.report.description)
+        reports = response.context_data["object_list"]
+        self.assertTrue(self.report in reports)
+
+    def test_custom_list_view_should_not_display_other_users_reports(self):
+        other_user = UserFactory()
+        other_report = ReportFactory(author=other_user, project=self.report.project)
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 200)
+        self.assertNotContains(response, other_report.description)
+
+    def test_custom_report_list_view_should_add_new_report_on_post(self):
+        response = self.client.post(self.url, self.data)
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(Report.objects.all().count(), 2)
+
+    def test_custom_report_list_view_should_add_user_to_project_selected_in_project_join_form_on_join(self):
+        new_project = ProjectFactory(name="New Project", start_date=datetime.datetime.now())
+        response = self.client.post(path=self.url, data={"projects": new_project.id, "join": "join"})
+        self.assertEqual(response.status_code, 302)
+        self.assertTrue(self.user in new_project.members.all())
+
+    def test_custom_report_list_view_should_not_add_user_to_project_selected_in_project_join_form_on_post(self):
+        new_project = ProjectFactory(name="New Project", start_date=datetime.datetime.now())
+        response = self.client.post(path=self.url, data={"projects": new_project.id})
+        self.assertEqual(response.status_code, 200)
+        self.assertFalse(self.user in new_project.members.all())
+
+    def test_custom_report_list_view_should_handle_no_project_being_selected_in_project_form_on_post(self):
+        new_project = ProjectFactory(name="New Project", start_date=datetime.datetime.now())
+        response = self.client.post(path=self.url, data={"join": "join"})
+        self.assertEqual(response.status_code, 200)
+        self.assertFalse(self.user in new_project.members.all())
+
+    def test_custom_report_list_view_should_dipslay_message_if_there_are_no_projects_available_to_join_to(self):
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, str(ReportListStrings.NO_PROJECTS_TO_JOIN.value))
+
+    @skip
```

Comment:
why skip?

Reply:
Because of the reason which I have described on rocket and you have fixed in https://github.com/Code-Poets/sheetstorm/issues/321. I will add link to this issue to `skip` decorator so we will remember to check it then.

Reply:
Yes, sorry. I made the review first and only then I read your explanation (and failed to delete this comment) ;)

Reply:
Sure, no problem, I just thought you asked for explanation that should be put into code, and I have waited for either issue to be created, or agreed that we can remove test (we did the first).